### PR TITLE
feat: Add v3 support

### PR DESCRIPTION
**Pull request by godexsoft**
This is enabling API v3 in order to support forwarding for the fix introduced in https://github.com/XRPLF/rippled/pull/4924

### CODE REVIEW

File: src/rpc/common/APIVersion.hpp [hunk 1]
```diff
@@ -34,16 +34,13 @@ static constexpr uint32_t API_VERSION_DEFAULT = 1u;
 
 /**
  * @brief Minimum API version supported by this build
- *
- * Note: Clio does not natively support v1 and only supports v2 or newer.
```
Comment (godexsoft) [inline src/rpc/common/APIVersion.hpp:38]: This is outdated and not true.

File: src/data/CassandraBackend.hpp [hunk 2]
```diff
@@ -950,7 +950,7 @@ class BasicCassandraBackend : public BackendInterface {
         std::vector<Statement> statements;
         statements.reserve(data.size());
         for (auto [mptId, holder] : data)
-            statements.push_back(schema_->insertMPTHolder.bind(std::move(mptId), std::move(holder)));
+            statements.push_back(schema_->insertMPTHolder.bind(mptId, holder));
```
Comment (kuznetsss) [inline src/data/CassandraBackend.hpp]: I don't think it is a related change.

File: src/rpc/common/APIVersion.hpp [hunk 3]
```diff
@@ -34,16 +34,13 @@ static constexpr uint32_t API_VERSION_DEFAULT = 1u;
...
- * However, Clio will forward all v1 requests to rippled for backward compatibility.
  */
 static constexpr uint32_t API_VERSION_MIN = 1u;
 
 /**
  * @brief Maximum API version supported by this build
  */
-static constexpr uint32_t API_VERSION_MAX = 2u;
+static constexpr uint32_t API_VERSION_MAX = 3u;
```
Comment (kuznetsss) [inline src/rpc/common/APIVersion.hpp:43]: Maybe we should have a test about this?

File: src/data/CassandraBackend.hpp (hunk 2 above)
Comment (godexsoft) [reply src/data/CassandraBackend.hpp]: It's not. it's a drive-by fix as stated in the description

File: src/rpc/common/APIVersion.hpp (hunk 3 above)
Comment (godexsoft) [reply src/rpc/common/APIVersion.hpp:43]: What would the test do? in `rippled` they have some ASSERTS for this but i'm not sure what the value is

File: src/data/CassandraBackend.hpp (hunk 2 above)
Comment (godexsoft) [reply src/data/CassandraBackend.hpp]: Fine, i'll move it to the other branch i have which is `clang-tidy` related

File: src/rpc/common/APIVersion.hpp (hunk 3 above)
Comment (kuznetsss) [reply src/rpc/common/APIVersion.hpp:43]: Some handler fails when api version is not 1 or 2 and now 3.

**Comment by cindyyan317** [comment]
Shall we use the rippled's apiMaximumSupportedVersion in xrpl directly for max

File: tests/unit/web/RPCServerHandlerTests.cpp [hunk 4]
```diff
@@ -870,3 +828,84 @@ TEST_F(WebRPCServerHandlerTest, WsRequestNotJson)
...
+    (*handler)(request, session);
+
+    auto response = boost::json::parse(session->message);
+    EXPECT_TRUE(response.is_object());
+    EXPECT_TRUE(response.as_object().contains("error"));
+    EXPECT_EQ(response.at("error").as_string(), "invalid_API_version");
+    EXPECT_TRUE(response.as_object().contains("error_message"));
+    EXPECT_EQ(response.at("error_message").as_string(), GetParam().wsMessage);
+    EXPECT_TRUE(response.as_object().contains("error_code"));
+    EXPECT_EQ(response.at("error_code").as_int64(), 6000);
```
Comment (kuznetsss) [inline tests/unit/web/RPCServerHandlerTests.cpp]: You could use the enum value of the error here.

**Review by kuznetsss — DISMISSED** [review]
LGTM

**Comment by godexsoft** [comment]
> apiMaximumSupportedVersion

@cindyyan317 I checked and I don't think we can use it. `rippled` sets it to 2 while we want 3. They have a concept of beta version but we don't.

**Review by cindyyan317 — DISMISSED** [review]

**Review by kuznetsss — APPROVED** [review]

**Pull request merged**